### PR TITLE
Update the permissions IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-03-05 15:57:55
Finished: 2018-03-05 16:13:37


<h2>Failing Jobs</h2>
<ul>

<li>safari:11.0</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25250)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/349346904)

**Comment by Honry** [comment]
Dup of https://github.com/w3c/web-platform-tests/pull/9301/

**Comment by lukebjerring** [comment]
@Honry - this PR trivially pulls https://github.com/tidoust/reffy-reports/blob/master/whatwg/idl/permissions.idl 

Could you append the header in that to your PR (https://github.com/web-platform-tests/wpt/pull/9301), so the next update is a no-op?

**Comment by Honry** [comment]
@lukebjerring, we've landed #9301, I think we can close it now. :)

**Comment by lukebjerring** [comment]
@Honry - reopened because I used the same local branch (now up to date after #9301)

This change to the test fixes the current issue in all non-chrome browsers:
https://label-param-dot-wptdashboard-staging.appspot.com/results/permissions/interfaces.any.worker.html?sha=d093f88ec3

File: permissions/interfaces.any.js [hunk 1]
```diff
@@ -11,18 +11,20 @@ if (self.importScripts) {
 // https://w3c.github.io/permissions/#idl-index
 
 promise_test(async () => {
+  const idl = await fetch("/interfaces/permissions.idl").then(r => r.text());
+  const dom = await fetch("/interfaces/dom.idl").then(r => r.text());
```
Comment (marcoscaceres) [inline permissions/interfaces.any.js]: Nit:... these can sometimes contain things that some browsers don't support... like `AbortController`. This means that tests fail, but for the wrong reasons. 

I don't know that the right approach is here...

**Review by marcoscaceres — APPROVED** [review]

**Pull request merged**